tempesta_fw/http.c

```diff
@@ -742,6 +743,53 @@ tfw_http_error_resp_switch(TfwHttpReq *req, int status, const char *reason)
 	tfw_http_send_resp(req, code);
 }
 
+static inline void
+tfw_http_srv_hmonitor(TfwHttpResp *resp)
+{
+	u32 crc32;
+	char *body_cstr;
+	TfwServer *srv = (TfwServer *)resp->conn->peer;
+
+	if (!test_bit(TFW_SRV_B_HMONITOR, (unsigned long *)&srv->hm_flags))
+		return;
+
+	if (tfw_apm_hm_srv_limit(resp->status, srv->apmref)) {
+		if (!tfw_srv_suspended(srv)) {
+			tfw_srv_mark_suspended(srv);
+			TFW_WARN_ADDR("server has been suspended: limit"
+				      " for bad responses is exceeded",
+				      &srv->addr);
+		}
+		return;
+	}
+
+	if (!tfw_srv_suspended(srv))
+		return;
+
+	if (!tfw_apm_hm_srv_status_alive(resp->status, srv->apmref)) {
```

Comment:
It's an extra validation, but perhaps it makes sense to check that the response has plain (not chunked) body. Just in case, as the body is then used in crc32 calculation.